### PR TITLE
Jonmv/simplify compile version selection

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review ü

### CODE REVIEW

**Comment by jonmv** [comment]
Ah, the change to prefer the new major for new apps snook into the second commit.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java [hunk 1]
```diff
@@ -269,17 +269,6 @@ public void writeUpgradesPerMinute(double n) {
         curator.set(upgradesPerMinutePath(), ByteBuffer.allocate(Double.BYTES).putDouble(n).array());
     }
 
-    public Optional<Integer> readTargetMajorVersion() {
-        return read(targetMajorVersionPath(), ByteBuffer::wrap).map(ByteBuffer::getInt);
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java:273]: Add a TODO for removing this path.

**Review by mpolden — APPROVED** [review]
Great!

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java:273]: Just forgot to remove it, fixed now :)

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java:273]: https://github.com/vespa-engine/vespa/pull/23930

**Pull request merged**